depmapomics/copynumbers.py

```diff
@@ -616,8 +618,206 @@ def read_ms_repeats(
+                    "START",
+                    "END",
+                    "LOG2_COPY_RATIO_POSTERIOR_50",
+                    "NUM_POINTS_COPY_RATIO",
+                ],
+                dtype={
+                    "CONTIG": "string",
+                    "START": "int64",
+                    "END": "int64",
+                    "LOG2_COPY_RATIO_POSTERIOR_50": "float64",
+                    "NUM_POINTS_COPY_RATIO": "int64",
+                },
+            ).assign(Sample=sample_id)
+        )
+
+    print("Aggregating segments")
+    segments = pd.concat(seg_dfs, ignore_index=True)
+    del seg_dfs
+
+    # exponentiate the log2 relative CNs
+    segments["LOG2_COPY_RATIO_POSTERIOR_50"] = (
+        2 ** segments["LOG2_COPY_RATIO_POSTERIOR_50"]
+    )
+
+    # segmentation algorithm doesn't output a CALL/Status column, so indicate missingness
+    segments["CALL"] = "."
+
+    # fix new columns' dtypes and column order
+    segments[["Sample", "CALL"]] = segments[["Sample", "CALL"]].astype("string")
+    seg_sample_ids = segments.pop("Sample")
+    segments.insert(loc=0, column="Sample", value=seg_sample_ids)
+
+    # sort and rename
+    segments = segments.sort_values(by=["Sample", "CONTIG", "START"], key=natsort_key)
+    segments = segments.rename(columns=constants.CNV_HMM_RENAMING)
+
+    # aggregate gene-level matrix
+    hgnc_table = make_hgnc_table()
+
+    print("Downloading gene copy number files")
+    gene_cn_files = cnv_files[gene_level_colname].dropna()  # [:100]
+    gene_dfs = []
+
+    for sample_id, f in tqdm(gene_cn_files.items(), total=len(gene_cn_files)):
+        gene_cn = pd.read_csv(
+            f,
+            sep="\t",
+            header=None,
+            names=[
+                "chr",
+                "start",
+                "end",
+                "annot",
+                "gene_body_len",
+                "cn_sum_weighted",
+                "num_probes",
+                "log2_rel_cn",
+            ],
+            usecols=["chr", "annot", "log2_rel_cn"],
+            dtype={"chr": "string", "annot": "string", "log2_rel_cn": "float64"},
+        ).assign(sample_id=sample_id)
+
+        gene_dfs.append(gene_cn)
+
+    print("Aggregating gene CNs")
+    gene_cn = pd.concat(gene_dfs, ignore_index=True)
+    del gene_dfs
+
+    # extract the Ensembl ID without suffixes
+    gene_cn["ensembl_gene_id"] = gene_cn["annot"].str.extract(r"(ENSG[0-9]+)")
+    gene_cn = gene_cn.drop(columns="annot")
+
+    # join the Hugo+Entrez ID column
+    gene_cn = gene_cn.merge(hgnc_table, how="inner", on="ensembl_gene_id")
+
+    # remove the PAR chrY CNs, keeping the chrX versions only
+    gene_cn = gene_cn.loc[~(gene_cn["chr"].eq("chrY") & gene_cn["par"])]
+    gene_cn = gene_cn.drop(columns=["chr", "ensembl_gene_id", "par"])
+
+    # exponentiate the log2 relative CNs
+    gene_cn["log2_rel_cn"] = 2 ** gene_cn["log2_rel_cn"]
+
+    # convert to wide format (sample by gene)
+    gene_cn_wide = gene_cn.pivot(
```

Comment:
The column names in gene-level CN from WES out of postProcess() are ENSG, and after merging with WGS they are renamed to hugo_entrez in
https://github.com/broadinstitute/depmap_omics/blob/25q2-cn-postprocessing/depmapomics/dm_omics.py#L742. So here can we pivot it to ensembl_gene_id in the output instead?

Reply:
and cnPostProcessing can call make_hgnc_table() instead of what is has now